runtime/compiler/runtime/JITClientSession.cpp

```diff
@@ -547,6 +552,22 @@ ClientSessionData::clearCaches()
       }
 
    _wellKnownClasses.clear();
+   setCachesAreCleared(true);
+   }
+
+void
+ClientSessionData::clearCachesLocked(TR_J9VMBase *fe)
+   {
+   // Acquire all required locks to clear client session caches
+   // This should be used when caches need to be cleared
+   // while the client is still sending requests
+   getSequencingMonitor()->enter();
+   writeAcquireClassUnloadRWMutex();
+   OMR::CriticalSection processUnloadedClasses(getROMMapMonitor());
```

Comment:
I would prefer the locks to be released in the reverse order of acquiring, thus we need to start a new block for the ROMMapMonitor:
```
{
OMR::CriticalSection processUnloadedClasses(getROMMapMonitor());
clearCaches(true);
}
```